subsys/net/ip/tcp2.c

```diff
@@ -1572,8 +1613,11 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
 	switch (conn->state) {
 	case TCP_LISTEN:
 		if (FL(&fl, ==, SYN)) {
+			/* Make sure our MSS is also send in the ACK */
+			conn->send_options.mss_found = true;
```

Comment:
Another concern is that reusing `struct tcp_options`, initially intended just for received options, also for send options, leads to a weird terminology, if using existing field names. IMHO, the fields should be renamed to more neutral first, e.g. `mss_found` -> `has_mss`.

(That's of course assuming the implementation approach taken in this PR is in the right direction.)